src/platform/stm32f1/hal/ticks.c

```diff
@@ -0,0 +1,32 @@
+/*
+ * SPDX-License-Identifier: MIT
+ * SPDX-FileCopyrightText: 2021 Rafael Silva <silvagracarafael@gmail.com>
+ */
+
+#include "platform/stm32f1/hal/ticks.h"
+#include "platform/stm32f1/systick.h"
+
+struct ticks_hal_t ticks_hal_init()
+{
+	struct ticks_hal_t hal = {
+		.ticks_ms = ticks_hal_ticks_ms,
+		.delay_ms = ticks_hal_delay_ms,
+		.delay_us = ticks_hal_delay_us,
+	};
+	return hal;
+}
+
+u64 ticks_hal_ticks_ms()
+{
+	return systick_get_ticks();
+}
+
+void ticks_hal_delay_ms(u32 ticks)
+{
+	return delay_ms(ticks);
+}
+
+void ticks_hal_delay_us(u32 ticks)
+{
+	return delay_us(ticks);
+}
```

Comment:
Here we don't really need helper functions as `systick_get_ticks`, `delay_ms` and `delay_us` seem to already have the interface we need.

These functions should also not be exported in the header. I see that I did it in the UHID platform, but that is wrong. I will open a PR fixing this.

Reply:
right, makes sense